packages/shared/lib/utils.ts

```diff
@@ -476,3 +476,12 @@ export function range(size: number, start: number = 0): number[] {
 export function getMonthYear(date: Date): string {
     return formatDate(date, { year: 'numeric', month: 'short' })
 }
+
+/*
+ * Gets an object value from an object string path
+ */
+export function resolveObjectPath(object: unknown, path: string, defaultValue: unknown): unknown {
+    if (!path) return
+
+    return path.split('.').reduce((o, p) => (o ? o[p] : defaultValue), object)
```

Comment:
I think users of this function might expect behavior similar to `lodash.get`, which resolves the whole path and then returns `defaultValue` if the value at the path is `undefined`. This seems like it will replace an intermediate object during a `reduce` operation with the `defaultValue`, which may be incorrect behavior in some cases. For example, in this [JSFiddle](https://jsfiddle.net/mjbqL29d/1/), `undefined` is returned instead of the `defaultValue`. Based on this [plain JS version](https://youmightnotneed.com/lodash#get), I think this should work:

```suggestion
    const result = path.split('.').reduce((o, p) => (o && o[p]), object)
    return result === undefined ? defaultValue : result
```